diff --git .gitignore .gitignore
index 4e9ff77..279d3f5 100644
--- .gitignore
+++ .gitignore
@@ -21,6 +21,7 @@ Docs/INFO_SRC
 Makefile
 TAGS
 Testing/
+tmp/
 VERSION.dep
 configure
 client/async_example
@@ -59,6 +60,9 @@ include/mysql_version.h
 include/mysqld_ername.h
 include/mysqld_error.h
 include/sql_state.h
+include/probes_mysql.d
+include/probes_mysql_dtrace.h
+include/probes_mysql_nodtrace.h
 info_macros.cmake
 libmysql*/libmysql*_exports_file.cc
 libmysql*/merge_archives_mysql*.cmake
@@ -68,6 +72,7 @@ libmysqld/examples/mysql_client_test_embedded
 libmysqld/examples/mysql_embedded
 libmysqld/examples/mysqltest_embedded
 make_dist.cmake
+mariadb-*.*.*.tar.gz
 mariadb-*.*.*/
 mysql-test/lib/My/SafeProcess/my_safe_process
 mysql-test/mtr
@@ -111,6 +116,8 @@ scripts/wsrep_sst_mysqldump
 scripts/wsrep_sst_rsync
 scripts/wsrep_sst_xtrabackup
 scripts/wsrep_sst_xtrabackup-v2
+scripts/maria_add_gis_sp.sql
+scripts/maria_add_gis_sp_bootstrap.sql
 sql-bench/bench-count-distinct
 sql-bench/bench-init.pl
 sql-bench/compare-results
@@ -165,32 +172,32 @@ storage/myisam/myisamlog
 storage/myisam/myisampack
 storage/myisam/rt_test
 storage/myisam/sp_test
-storage/tokudb/ft-index/buildheader/db.h
-storage/tokudb/ft-index/buildheader/make_tdb
-storage/tokudb/ft-index/buildheader/runcat.sh
-storage/tokudb/ft-index/ft/log_code.cc
-storage/tokudb/ft-index/ft/log_header.h
-storage/tokudb/ft-index/ft/log_print.cc
-storage/tokudb/ft-index/ft/logformat
-storage/tokudb/ft-index/ft/ftverify
-storage/tokudb/ft-index/ft/tdb-recover
-storage/tokudb/ft-index/ft/tokuftdump
-storage/tokudb/ft-index/portability/merge_archives_tokuportability_static.cmake
-storage/tokudb/ft-index/portability/toku_config.h
-storage/tokudb/ft-index/portability/tokuportability_static_depends.cc
-storage/tokudb/ft-index/src/merge_archives_tokufractaltree_static.cmake
-storage/tokudb/ft-index/src/tokufractaltree_static_depends.cc
-storage/tokudb/ft-index/toku_include/toku_config.h
-storage/tokudb/ft-index/tools/ba_replay
-storage/tokudb/ft-index/tools/ftverify
-storage/tokudb/ft-index/tools/tdb-recover
-storage/tokudb/ft-index/tools/tdb_logprint
-storage/tokudb/ft-index/tools/tokudb_dump
-storage/tokudb/ft-index/tools/tokudb_gen
-storage/tokudb/ft-index/tools/tokudb_load
-storage/tokudb/ft-index/tools/tokuftdump
-storage/tokudb/ft-index/tools/tokuft_logprint
-storage/tokudb/ft-index/xz/
+storage/tokudb/PerconaFT/buildheader/db.h
+storage/tokudb/PerconaFT/buildheader/make_tdb
+storage/tokudb/PerconaFT/buildheader/runcat.sh
+storage/tokudb/PerconaFT/ft/log_code.cc
+storage/tokudb/PerconaFT/ft/log_header.h
+storage/tokudb/PerconaFT/ft/log_print.cc
+storage/tokudb/PerconaFT/ft/logformat
+storage/tokudb/PerconaFT/ft/ftverify
+storage/tokudb/PerconaFT/ft/tdb-recover
+storage/tokudb/PerconaFT/ft/tokuftdump
+storage/tokudb/PerconaFT/portability/merge_archives_tokuportability_static.cmake
+storage/tokudb/PerconaFT/portability/toku_config.h
+storage/tokudb/PerconaFT/portability/tokuportability_static_depends.cc
+storage/tokudb/PerconaFT/snappy/
+storage/tokudb/PerconaFT/src/merge_archives_tokufractaltree_static.cmake
+storage/tokudb/PerconaFT/src/tokufractaltree_static_depends.cc
+storage/tokudb/PerconaFT/toku_include/toku_config.h
+storage/tokudb/PerconaFT/tools/ba_replay
+storage/tokudb/PerconaFT/tools/ftverify
+storage/tokudb/PerconaFT/tools/tdb-recover
+storage/tokudb/PerconaFT/tools/tokudb_dump
+storage/tokudb/PerconaFT/tools/tokudb_gen
+storage/tokudb/PerconaFT/tools/tokudb_load
+storage/tokudb/PerconaFT/tools/tokuftdump
+storage/tokudb/PerconaFT/tools/tokuft_logprint
+storage/tokudb/PerconaFT/xz/
 support-files/MySQL-shared-compat.spec
 support-files/binary-configure
 support-files/config.huge.ini
@@ -237,6 +244,8 @@ storage/mroonga/mysql-test/mroonga/storage/r/variable_version.result
 *.exp
 *.dep
 *.idb
+*.res
+*.tlog
 
 # Precompiled Headers
 *.gch
diff --git patches/el5-asm-compat.patch patches/el5-asm-compat.patch
deleted file mode 100644
index 1e3f2a7..0000000
--- patches/el5-asm-compat.patch
+++ /dev/null
@@ -1,15 +0,0 @@
-diff -ru mariadb-10.1.9.orig/include/my_context.h mariadb-10.1.9/include/my_context.h
---- mariadb-10.1.9.orig/include/my_context.h   2015-11-20 17:08:02.000000000 -0800
-+++ mariadb-10.1.9/include/my_context.h        2016-01-27 00:26:50.000000000 -0800
-@@ -27,8 +27,10 @@
- 
- #ifdef __WIN__
- #define MY_CONTEXT_USE_WIN32_FIBERS 1
--#elif defined(__GNUC__) && __GNUC__ >= 3 && defined(__x86_64__) && !defined(__ILP32__)
-+#elif defined(__clang__)
- #define MY_CONTEXT_USE_X86_64_GCC_ASM
-+#elif defined(__GNUC__) && __GNUC__ >= 3 && defined(__x86_64__) && !defined(__ILP32__)
-+#define MY_CONTEXT_USE_UCONTEXT
- #elif defined(__GNUC__) && __GNUC__ >= 3 && defined(__i386__)
- #define MY_CONTEXT_USE_I386_GCC_ASM
- #elif defined(HAVE_UCONTEXT)
